src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,92 +48,1243 @@
+                                    Map.Entry<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesFromRecipient) {
+        return getCurrentSection(responsesFromRecipient, false);
+    }
+    
+
+    /**
+     * Uses the first response to get the current section
+     * @param responses
+     */
+    private String getCurrentSection(
+                        Map.Entry<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responses,
+                        boolean isGiver) {
+        String currentSection = Const.DEFAULT_SECTION;
+        // update current section
+        // retrieve section from the first response of this user
+        // TODO simplify by introducing more data structures into bundle
+        for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> responsesFromGiverForQuestion : 
+                                                                                     responses.getValue().entrySet()) {
+            if (responsesFromGiverForQuestion.getValue().isEmpty()) {
+                continue;
+            }
+            FeedbackResponseAttributes firstResponse = responsesFromGiverForQuestion.getValue().get(0);
+            currentSection = isGiver ? firstResponse.giverSection : firstResponse.recipientSection ;
+            break;
+        }
+        
+        return currentSection;
+    }
+
+    private void createMissingParticipantPanelsWithModerationButtonForPrevTeam(
+                                            InstructorFeedbackResultsSectionPanel sectionPanel, String prevTeam,
+                                            Set<String> teamsWithResponses, Set<String> teamMembersWithResponses,
+                                            boolean isGiver) {
+        createMissingParticipantPanelsForPrevTeam(sectionPanel, prevTeam, teamsWithResponses, 
+                                                  teamMembersWithResponses, true, isGiver);
+    }
+    
+    private void createMissingParticipantPanelsWithoutModerationButtonForPrevTeam(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, String prevTeam,
+                                    Set<String> teamsWithResponses, Set<String> teamMembersWithResponses,
+                                    boolean isGiver) {
+        createMissingParticipantPanelsForPrevTeam(sectionPanel, prevTeam, teamsWithResponses, 
+                                                  teamMembersWithResponses, false, isGiver);
+    }    
+    
+    private void createMissingParticipantPanelsForPrevTeam(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, String prevTeam,
+                                    Set<String> teamsWithResponses, Set<String> teamMembersWithResponses,
+                                    boolean isDisplayingModerationButton, boolean isGiver) {
+        boolean isFirstTeam = prevTeam.isEmpty();
+        if (!isFirstTeam) {
```

Comment:
Maybe this instead? It should be neater than placing nearly everything in this function in an if-block.
```
if (isFirstTeam) {
    return;
}
```